examples/discovery-search-app/src/ExampleDocumentProvider.js

```diff
@@ -0,0 +1,24 @@
+import { DocumentProvider } from '@ibm-watson/discovery-react-components';
+
+/**
+ * This is an example implementation of the `DocumentProvider` interface. This will return the PDF
+ * contents for the "Art Effects" doc.
+ *
+ * To see this in action, create a new collection in your project and upload
+ * "/examples/discovery-search-app/public/documents/Art Effects.pdf". Then, when running the
+ * example app against this collection, if you search and find this document, when you click
+ * "View passage in document" (or similar) to open the document view, you will see the PDF
+ * itself rendered, instead of HTML or text.
+ */
+export class ExampleDocumentProvider implements DocumentProvider {
+  async provides(document: DocumentProviderProps) {
+    return document.extracted_metadata.filename === 'Art Effects.pdf';
```

Comment:
🐛 One issue here is how will consumers of this library know which file to load? There exists a unique `document_id` for each document, but that's specific to Discovery and means nothing to end users.

Ideally, a consumer would be able to associate their own ID with the uploaded document, which Disco could then store with the document.

Any other ideas on how to handle this?

Reply:
🤔